0003-lightning-as-alpha-ledger.md

```diff
@@ -0,0 +1,158 @@
+
+| Name                    | JSON Encoding | Description                                                                                               |
+|-------------------------|---------------|-----------------------------------------------------------------------------------------------------------|
+| `alpha_redeem_identity` | `α::Identity` | The identity on alpha that **A**, this is the LN `invoice`  |
+| `beta_refund_identity`  | `β::Identity` | The identity on beta that **B** will be transferred to when the beta-HTLC is activated after `beta_expiry`      |
+
+4) Alice now starts the [exection phase](https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md#1-alice-deploys-%CE%B1-htlc) by paying the invoice using the LND command `sendpayment`.
+5) Bob gets notified about funding of alpha (i.e. the invoice has been paid but cannot be settled yet), and continues with [deploying beta-HTLC](https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-Basic.md#2-bob-deploys-%CE%B2-htlc), i.e. he deploys a HTLC on Ethereum.
+6) As soon as beta has enough confirmations for Alice, she redeems the beta-HTLC using her secret.
+7) Bob gets notified about this, learns the secret and can now settle the LND invoice by invoking the LND command `settleInvoice` ([more details](https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L53)).
+
+The trade is complete.
+
+### Considerations and limitations
+
+#### Lightning Network: a new ledger
+Similiar to the ledger definitions for [Bitcoin](https://github.com/comit-network/RFCs/blob/master/RFC-004-Bitcoin.md) and [Ethereum](https://github.com/comit-network/RFCs/blob/master/RFC-006-Ethereum.md) we need to handle the Lightning Network differently.
+This is required because the comit-node and btsieve need to perform different actions accordingly.
+We are always talking about Ledgers and Assets, (e.g. _Bitcoin_ Asset on the _Bitcoin_ Ledger, _Ether_ Asset on the _Ethereum_ Ledger, _Erc20_ on the _Ethereum_ Ledger, ...), Hence,
+if we follow this approach, for supporting LN (through LND) we will need to introduce a new pair of Ledger and Asset:
+
+* Ledger: the **Lightning Network**. _Ledgers_ are used as _settlement layers_ for our HTLCs. In the case of LND, this layer is the Lightning Network.
+* Asset: **Bitcoin**. Since LN is a layer-2 network on top of Bitcoin, the asset should also be Bitcoin.
+
+#### Timeouts
+When calling `addholdinvoice` an [`expiry`](https://github.com/lightningnetwork/lnd/blob/aa1cd04dbf07a9195d5ada752f383988d8d01fa7/cmd/lncli/invoicesrpc_active.go#L142) (in seconds) can be given. However, this timeout is relative and will get accumulated across the payment path.
+Hence, Alice will need to know the path and this value in advance before sending a swap request to Bob as she won't (or should not) be able to change neither `alpha_expiry` nor `beta_expiry` after the swap request.
```

Comment:
This is a combination of parameters that are configured globally for the payee node and I think on-the-go (when paying) for the payer node. Need to dig in my notes.

The payer configures with `--final_cltv_delta=T` when paying and the payee configures when opening a channel or global conf for the lnd.

It means that the payee needs to accept/propose an expiry that matches their lnd conf.

It means that the payer could deviate from the agreed expiry and lnd may:
- accept the invoice if it is still above the configure threshold (CSV I think)
- reject the invoice automatically if it is under the threshold.

but then, the HTLC details are visible to the payee so the payee could decide not to proceed with the swap if the expiry does not match expectations

Reply:
Thanks for pointing this out.

Reply:
In the case of LN being alpha ledger then it's fine: It means that the payee needs to accept~/propose~ an expiry that matches their lnd conf.